task4/VCS/Watcher.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace VCS
+{
+    /// <summary>
+    /// A class that monitors the file system
+    /// </summary>
+    public class Watcher
+    {
+        private DateTime lastChange;
+        private FileSystemWatcher watcher;
+
+        public Action<string> ChangeHandler;
+        public Commit Commit { get; private set; }
+        
+        public Watcher()
+        {
+            watcher = new FileSystemWatcher
+            {
+                Path = Storage.DirectoryToWatch,
+                Filter = "*.txt",
+                IncludeSubdirectories = true,
+                NotifyFilter = NotifyFilters.FileName |
+                NotifyFilters.LastWrite |
+                NotifyFilters.CreationTime,
+                EnableRaisingEvents = true,
+            };
+            watcher.Changed += OnChanged;
+            watcher.Created += OnCreated;
+            watcher.Deleted += OnDeleted;
+            watcher.Renamed += OnRenamed;
+            Commit = new Commit(new List<Change>());
+            lastChange = DateTime.MinValue;
+        }
+
+        #region EventHandlers
+        private void OnChanged(object source,FileSystemEventArgs file)
+        {
+            var lastWriteTime = File.GetLastWriteTime(file.FullPath);
+            if (lastWriteTime != lastChange)
+            {
+                ChangeHandler?.Invoke($"File: {file.FullPath} {file.ChangeType}");
+                lastChange = lastWriteTime;
+                Commit.Changes.Add(new Change(file.FullPath, file.ChangeType, lastWriteTime));
+            }
```

Comment:
* And if 2 or more files are changed at the same time?
* Changed event is also raised on rename, create or delete